### PR TITLE
chore: update to `nightly-2024-08-20`

### PR DESCRIPTION
**Pull request by Tropix126**
## Describe the changes this PR makes. Why should it be merged?
Updates vexide's internal toolchain from `nightly-2024-02-07` to `nightly-2024-08-20` and fixes associated lints.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@
           buildInputs = [
             cargo-v5'
             pkgs.cargo-binutils
-            (pkgs.rust-bin.nightly."2024-02-07".default.override {
+            (pkgs.rust-bin.nightly.default.override {
```
Comment (Gavin-Niederman) [inline flake.nix:21]: this needs to be pkgs.rust-bin.nightly.latest.default.override

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]

File: packages/vexide-devices/src/adi/solenoid.rs [hunk 2]
```diff
@@ -65,9 +65,9 @@ impl AdiSolenoid {
     /// Close the solenoid.
     ///
     /// - On single-acting solenoids (e.g. SY113-SMO-PM3-F), this will simply block air pressure
-    /// through the "open" valve.
+    ///   through the "open" valve.
     /// - On double-acting solenoids (e.g. SYJ3120-SMO-M3-F), this will block air pressure through
-    /// the "open" valve and allow air pressure into the "close" valve.
+    ///   the "open" valve and allow air pressure into the "close" valve.
```
Comment (Gavin-Niederman) [inline packages/vexide-devices/src/adi/solenoid.rs:70]: These have two spaces

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]

File: packages/vexide-devices/src/adi/solenoid.rs (hunk 2 above)
Comment (Tropix126) [reply packages/vexide-devices/src/adi/solenoid.rs:70]: This is intentional. It lines up the start of the text with the text on the first line, and clippy will yell at us if we don't do this.
image

**Review by Gavin-Niederman — APPROVED** [review]
LGTM :3

**Pull request merged**